src/pages/global-authorization/forms/FieldGlobalForm.js

```diff
@@ -0,0 +1,216 @@
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { user, setUser } = useContext(AuthContext)
+  const [isLoading, setIsLoading] = useState(false)
+
+  const invalidate = useInvalidate({
+    endpointId: SystemRepository.ModuleClassRES.qry
+  })
+
+  const detailsFormik = useFormik({
+    enableReinitialize: true,
+    validateOnChange: true,
+    initialValues: {
+      rows: [],
+      resourceId: resourceId,
+      resourceName: resourceName
+    },
+    onSubmit: async obj => {
+
+      const updatedRows = detailsFormik.values.rows
+        .filter(obj => {
+          return obj.accessLevel != null
+        }).map(({ ...rest }) => ({
+          resourceId: resourceId,
+          ...rest
+        }))
+
+      // Create the resultObject
+      const resultObject = {
+        resourceId: resourceId,
+        controls: updatedRows
+      }
+
+      const response = await postRequest({
+        extension: AccessControlRepository.GlobalControlAuthorizationPack.set2,
+        record: JSON.stringify(resultObject)
+      })
+
+      toast.success('Record Edited Successfully')
+
+      invalidate()
+    }
+  })
+
+  const columns = [
+    {
+      component: 'textfield',
+      label: labels.controlId,
+      name: 'controlId',
+      props: {
+        readOnly: true
+      }
+    },
+    {
+      component: 'textfield',
+      label: labels.controlName,
+      name: 'name',
+      props: {
+        readOnly: true
+      }
+    },
+    {
+      component: 'resourcecombobox',
+      name: 'accessLevelCombo',
+      label: labels.accessLevel,
+      props: {
+        datasetId: DataSets.AU_RESOURCE_CONTROL_ACCESS_LEVEL,
+        valueField: 'key',
+        displayField: 'value',
+        mapping: [ { from: 'key', to: 'accessLevel' } ,
+          { from: 'value', to: 'accessLevelName' }]
+      },
+      async onChange({ row: { update, newRow } }) {
+        if (!newRow?.accessLevelCombo?.key) {
+          return
+        } else {
+          update({ accessLevelName: newRow.accessLevelCombo?.value, accessLevel: newRow.accessLevelCombo?.key })
+        }
+      }
+    }
+  ]
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        if (resourceId) {
+          setIsLoading(true)
+
+          const res = await getRequest({
+            extension: SystemRepository.ResourceControl.qry,
+            parameters: `_resourceId=${resourceId}`
+          })
+
+          const accessLevelRes = await getRequest({
+            extension: AccessControlRepository.GlobalControlAuthorizationView.qry,
+            parameters: `_resourceId=${resourceId}`
+          })
+
+          const finalList = res.list.map(controlDetail => {
+            const n = {
+              controlId: controlDetail.id,
+              name: controlDetail.name,
+              accessLevel: null,
+              accessLevelName: null
+            };
+    
+            const matching = accessLevelRes.list.find(
```

Comment:
Sounds like confusing logic to me here. Can we get a brief explanation on why should we do such a mapping?

Reply:
here we fill the grid from 2 apis.. As if we have a class with 4 fields each one from a separate api so the first request we get the 2 fields we need and change their names to suit our grid binding and post.. Then we compare the id and get the rest of info from second request.